.github/workflows/pipeline.yaml

```diff
@@ -0,0 +1,76 @@
+name: Azure Pipeline Merge Queue Check
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+    - master
+  merge_group:
+    types:
+      - checks_requested
+jobs:
+  Azure-Merge-Queue-Check:
+    if: ${{ github.event_name == 'merge_group' }}
+    strategy:
+      matrix:
+        go-version: ['1.21']
+        os: [ubuntu-latest]
+    name: Azure Pipeline Merge Queue Check
+    runs-on: ${{ matrix.os }}
+    permissions:
+      actions: read
+      contents: read
+      id-token: write
+    steps:
+    - name: Azure Login
+      uses: Azure/login@v1.5.1
+      env:
+        AZURE_CORE_OUTPUT: none
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    - name: Check Azure Pipelines
+      uses: azure/CLI@v1
+      env:
+        AZURE_CORE_OUTPUT: none
+      with:
+        azcliversion: latest
+        inlineScript: |
+          az account show
+          echo ${{ secrets.AZURE_DEVOPS_EXT_PAT }} | az devops login --org ${{ secrets.AZURE_PIPELINE_ORG }}
+
+          echo "Sanity check recent Pipeline runs"
+          az pipelines runs list --pipeline-ids ${{ secrets.AZURE_PIPELINE_ID }} --org ${{ secrets.AZURE_PIPELINE_ORG }} --project ${{ secrets.AZURE_PIPELINE_PROJECT }} --reason individualCI --top 10 --output json | jq -r .[].sourceBranch
```

Comment:
What's the use of the call ?

Reply:
On the happy path where AZP and GH properly queue up a pipeline run before this workflow start, line 41's output will show the merge queue PR branch i.e. `refs/heads/gh-readonly-queue/master/pr-2460-****` as part of the 10 most recent `individualCI` runs. 

We can check `all, batchedCI, buildCompletion, checkInShelveset, individualCI, manual, pullRequest, schedule, triggered, userCreated, validateShelveset` for the reason of the queued run, but the only ones we deal with are `individualCI, manual, pullRequest`. 

Of note, how these runs will be triggered:
Happy path Merge Queue - individualCI
Workflow Check Run using az pipelines (line 60) - manual
Precheck from PR - pullRequest

This line is purely for debugging and adds additional context for the current status of pipeline.